### PR TITLE
RCDs can be upgraded to create computer/machine frames, moved deconstruction to the upgrade tier

### PR DESCRIPTION
**Pull request by Sishen1542**
## About The Pull Request
port of https://github.com/tgstation/tgstation/pull/42388
what it says on the tin

## Why It's Good For The Game
Makes griefing with an RCD much harder to do while not removing a feature entirely. Disks are easy and cheap to produce (Engi lathe only)

## Changelog
:cl: original by Tlaltecuhtli, port by sishen1542
add: rcd disk that gives rcd computer frame and machine frame designs
add: upgraded rcd in CE lockers
tweak: moved deconstruction to the upgrade disk, ashwalker rcd comes upgraded
/:cl:

### CODE REVIEW

**Comment by BurgerLUA** [comment]
I mean it's a good PR but it is no way a replacement for my other PR.

**Comment by Sishen1542** [comment]
made borg and admin RCDs upgraded by default

**Comment by terranaut1** [comment]
absolutely garbage, deconstruct rcd is a tool antags use and rely on and you're locking it behind tech nobody gets to be printed in a place that very territorial people lounge around in there's a reason other servers dont have it like that you goomba

**Comment by chefferz** [comment]
Really unbalanced PR that most engineer mains (like myself) rely on to quickly take care of emergency situations. Another one of those things that if a player is griefing with it, then ahelp them, don't remove a feature that some departments rely on in emergencies because of it.

**Comment by RogueSteampunker** [comment]
Once again, just ahelp the people abusing it, rather than nerf it into actual unuse. To start, the advanced engineering node is rarely ever researched, which means the chances of actually getting the disk are rare.

Second, the main issue with the decon is the fact that it tears down doors, where just making it take a good deal of time and a large clip to decon them would do much better. This just makes rcds actually worthless until science maybe perhaps researches the node for it. And at that point, you have jaws of life, and power tools. Why would you ever want to fuck with the rcd?

File: _maps/RandomRuins/LavaRuins/lavaland_surface_ash_walker1.dmm [hunk 1]
```diff
@@ -508,7 +508,9 @@
 /obj/structure/stone_tile/cracked{
 	dir = 4
 	},
-/obj/item/construction/rcd/loaded,
+/obj/item/construction/rcd/loaded{
+	upgrade = 1
+	},
```
Comment (deathride58) [inline _maps/RandomRuins/LavaRuins/lavaland_surface_ash_walker1.dmm:513]: Make this a subtype.

**Review by deathride58 — DISMISSED** [review]
This is a good PR tbh.
Also, please refrain from commenting on PRs if your only feedback is "jUST AHELP IF SOMEONE ABUSES IT!!!". This isn't Bay where 90% of the content in the game is hilariously broken.
Additionally, this is placed under the advanced engineering node, which is a node that's required fairly early on in the current RnD meta, meaning the RCD upgrade is more or less guaranteed to be available in the engineering techfab before the 10 minute mark so long as there's scientists that know what they're doing. Please actually do basic analysis of a PR before commenting.

**Comment by Sishen1542** [comment]
CE locker now gets a single upgraded RCD, 3 in vendor unchanged

**Review by nik707 — APPROVED** [review]

**Comment by nik707** [comment]
@deathride58

**Comment by Tupinambis** [comment]
Just make RCDs not printable in the autolathe. That's literally it. Please do not nerf engineering because everyone has access to RCDs now. This harms both traitor engineers and engineers that want to save time.

**Comment by Tupinambis** [comment]
If you really want to keep RCDs in the autolathes, at least KEEP deconstruction roundstart on engineer RCDs.

**Review by silicons — APPROVED** [review]

**Pull request merged**